static/js/stream_edit.js

```diff
@@ -960,8 +960,8 @@ export function initialize() {
         }
 
         function do_archive_stream() {
-            const stream_id = $(".confirm_dialog_yes_button").data("stream-id");
-            if (!stream_id) {
+            const dialog_stream_id = $(".confirm_dialog_yes_button").data("stream-id");
+            if (!dialog_stream_id) {
```

Comment:
I think in this case, it might be better to just remember the `stream_id` from the context rather than re-parsing it; that would let us delete the error handling code that makes up most of this function as well, leaving just the last 2 lines.